aws/resource_aws_wafv2_rule_group.go

```diff
@@ -0,0 +1,1346 @@
+		m["priority"] = int(aws.Int64Value(rule.Priority))
+		m["statement"] = flattenWafv2RootStatement(rule.Statement)
+		m["visibility_config"] = flattenWafv2VisibilityConfig(rule.VisibilityConfig)
+		out[i] = m
+	}
+
+	return out
+}
+
+func flattenWafv2RuleAction(a *wafv2.RuleAction) interface{} {
+	if a == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{}
+
+	if a.Allow != nil {
+		m["allow"] = make([]map[string]interface{}, 1)
+	}
+
+	if a.Block != nil {
+		m["block"] = make([]map[string]interface{}, 1)
+	}
+
+	if a.Count != nil {
+		m["count"] = make([]map[string]interface{}, 1)
+	}
+
+	return []interface{}{m}
+}
+
+func flattenWafv2RootStatement(s *wafv2.Statement) interface{} {
+	if s == nil {
+		return []interface{}{}
+	}
+
+	return []interface{}{flattenWafv2Statement(s)}
+}
+
+func flattenWafv2Statements(s []*wafv2.Statement) interface{} {
+	out := make([]interface{}, len(s))
+	for i, statement := range s {
+		out[i] = flattenWafv2Statement(statement)
+	}
+
+	return out
+}
+
+func flattenWafv2Statement(s *wafv2.Statement) map[string]interface{} {
+	if s == nil {
+		return map[string]interface{}{}
+	}
+
+	m := map[string]interface{}{}
+
+	if s.AndStatement != nil {
+		m["and_statement"] = flattenWafv2AndStatement(s.AndStatement)
+	}
+
+	if s.ByteMatchStatement != nil {
+		m["byte_match_statement"] = flattenWafv2ByteMatchStatement(s.ByteMatchStatement)
+	}
+
+	if s.IPSetReferenceStatement != nil {
+		m["ip_set_reference_statement"] = flattenWafv2IpSetReferenceStatement(s.IPSetReferenceStatement)
+	}
+
+	if s.GeoMatchStatement != nil {
+		m["geo_match_statement"] = flattenWafv2GeoMatchStatement(s.GeoMatchStatement)
+	}
+
+	if s.NotStatement != nil {
+		m["not_statement"] = flattenWafv2NotStatement(s.NotStatement)
+	}
+
+	if s.OrStatement != nil {
+		m["or_statement"] = flattenWafv2OrStatement(s.OrStatement)
+	}
+
+	if s.RegexPatternSetReferenceStatement != nil {
+		m["regex_pattern_set_reference_statement"] = flattenWafv2RegexPatternSetReferenceStatement(s.RegexPatternSetReferenceStatement)
+	}
+
+	if s.SizeConstraintStatement != nil {
+		m["size_constraint_statement"] = flattenWafv2SizeConstraintStatement(s.SizeConstraintStatement)
+	}
+
+	if s.SqliMatchStatement != nil {
+		m["sqli_match_statement"] = flattenWafv2SqliMatchStatement(s.SqliMatchStatement)
+	}
+
+	if s.XssMatchStatement != nil {
+		m["xss_match_statement"] = flattenWafv2XssMatchStatement(s.XssMatchStatement)
+	}
+
+	return m
+}
+
+func flattenWafv2AndStatement(a *wafv2.AndStatement) interface{} {
+	if a == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{
+		"statement": flattenWafv2Statements(a.Statements),
+	}
+
+	return []interface{}{m}
+}
+
+func flattenWafv2ByteMatchStatement(b *wafv2.ByteMatchStatement) interface{} {
+	if b == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{
+		"field_to_match":        flattenWafv2FieldToMatch(b.FieldToMatch),
+		"positional_constraint": b.PositionalConstraint,
```

Comment:
similar comment as below to consistently set the values in this map; positional constraint can be set with `aws.StringValue(b.PositionalConstraint)`